src/main/java/io/jenkins/plugins/checks/github/CheckRunGHEventSubscriber.java

```diff
@@ -77,19 +81,27 @@ protected void onEvent(final GHSubscriberEvent event) {
             return;
         }
 
+        try {
+            JSONObject payloadJSON = new JSONObject(payload);
+            branchName = payloadJSON.getJSONObject("check_run").getJSONObject("check_suite").getString("head_branch");
+        } 
+        catch  (JSONException e) {
+            throw new IllegalStateException("Could not parse check run event: " + payload.replaceAll("[\r\n]", ""), e);
+        }
+
         LOGGER.log(Level.INFO, "Received rerun request through GitHub checks API.");
         try (ACLContext ignored = ACL.as(ACL.SYSTEM)) {
-            scheduleRerun(checkRun);
+            scheduleRerun(checkRun, branchName);
```

Comment:
The actual code that does something useful would be simpler to find if you would use one try-catch block for the whole method (calls in line 73 and 86), then you can integrate the variable assignments in the declaration.

Reply:
I changed the first part of the code in https://github.com/jenkinsci/github-checks-plugin/pull/227.